sources/Afup/Association/Cotisations.php

```diff
@@ -384,21 +387,87 @@ function genererFacture($id_cotisation, $chemin = null)
         $pdf->Ln(15);
 
         $pdf->MultiCell(180, 5, utf8_decode("Facture concernant votre adhésion à l'Association Française des Utilisateurs de PHP (AFUP)."));
-        // Cadre
-        $pdf->Ln(10);
-        $pdf->SetFillColor(200, 200, 200);
-        $pdf->Cell(50, 5, 'Code', 1, 0, 'L', 1);
-        $pdf->Cell(100, 5, utf8_decode('Désignation'), 1, 0, 'L', 1);
-        $pdf->Cell(40, 5, 'Prix', 1, 0, 'L', 1);
 
-        $pdf->Ln();
-        $pdf->SetFillColor(255, 255, 255);
-        $pdf->Cell(50, 5, 'ADH', 1);
-        $pdf->Cell(100, 5, utf8_decode("Adhésion AFUP jusqu'au " . date('d/m/Y', $cotisation['date_fin'])), 1);
-        $pdf->Cell(40, 5, utf8_decode($cotisation['montant'] . ' '), 1);
+        if (false === $isSubjectedToVat) {
+            // Cadre
+            $pdf->Ln(10);
+            $pdf->SetFillColor(200, 200, 200);
+            $pdf->Cell(50, 5, 'Code', 1, 0, 'L', 1);
+            $pdf->Cell(100, 5, utf8_decode('Désignation'), 1, 0, 'L', 1);
+            $pdf->Cell(40, 5, 'Prix', 1, 0, 'L', 1);
+
+            $pdf->Ln();
+            $pdf->SetFillColor(255, 255, 255);
+            $pdf->Cell(50, 5, 'ADH', 1);
+            $pdf->Cell(100, 5, utf8_decode("Adhésion AFUP jusqu'au " . date('d/m/Y', $cotisation['date_fin'])), 1);
```

Comment:
"`utf8_decode()` est OBSOLÈTE à partir de PHP 8.2.0. Dépendre de cette fonction est fortement déconseillé. "
```suggestion
            $pdf->Cell(100, 5, mb_convert_encoding("Adhésion AFUP jusqu'au " . date('d/m/Y', $cotisation['date_fin']), 'ISO-8859-1', 'UTF-8'), 1);

```

Reply:
là dessus j'aurais tendance à faire le changement sur tous les endroits où on génère les PDFs en même temps pour assurer la cohérence.